packages/core/android/src/main/java/io/embrace/embracewrapper/EmbraceManagerModule.java

```diff
@@ -562,6 +568,110 @@ public void logNetworkClientError(String url,
+            stringMap.put(key, readableMap.getString(key));
+        }
+
+        return stringMap;
+    }
+
+    private ErrorCode getSpanErrorCodebyString(String errorCode){
+        switch (errorCode) {
+            case "Failure":
+                return ErrorCode.FAILURE;
+
+            case "UserAbandon":
+                return ErrorCode.USER_ABANDON;
+
+            case "Unknown":
+                return ErrorCode.UNKNOWN;
+
+            default:
+                return null;
+        }
+    }
+
+    private List<Map<String, Object>> transformListReadableMapToListMap(ReadableArray items) {
+        List<Map<String, Object>> objectMapList = new ArrayList<>();
+
+        try{
+            for (int i = 0; i < items.size(); i++) {
+                ReadableMap readableMap = items.getMap(i);
+
+                if (readableMap != null) {
+                    Map<String, Object> map = readableMap.toHashMap();
+                    // TODO Change when Android/iOS replace time in nano for ms
+                    if (map.containsKey("timestampNanos") && map.get("timestampNanos") instanceof Double) {
+                        double timestampNanos = (Double) map.get("timestampNanos");
+                        map.put("timestampNanos", (long) timestampNanos);
+                    }
+                    objectMapList.add(map);
+                }
+            }
+        }catch(Exception e){
+            Log.e("Embrace", "There was an error in parsing the span event data", e);
+        }
+
+        return objectMapList;
+    }
+
+    @ReactMethod()
+    public void stopSpan(String spanId, String errorCodeString, Promise promise) {
+        try{
+            
+            ErrorCode errorCodeInstance = this.getSpanErrorCodebyString(errorCodeString);
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().stopSpan(spanId, errorCodeInstance, null));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void addSpanAttributeToSpan(String spanId, String key, String value, Promise promise) {
+        try{
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().addSpanAttribute(spanId, key, value));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void addSpanEventToSpan(String spanId, String name, Double time, ReadableMap attributes, Promise promise) {
+        try{
+
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().addSpanEvent(spanId, name, time.longValue(), this.convertToReadableMap(attributes)));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void recordCompletedSpan(String name, Double startTimeNanos, Double endTimeNanos, String errorCodeString, String parentSpanId, ReadableMap attributes, ReadableArray events, Promise promise) {
```

Comment:
Since this is open sourced, you might want to add a comment here saying the Android SDK expects millisecond but this works for now because of the workaround.

Reply:
I will let it with that signature, since I need to match the parameters with iOS. In the next version , when iOS update the parameter, I will refactor this